nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/PathHelper.cs

```diff
@@ -111,4 +111,22 @@ public static void CopyDirectory(string sourceDirectory, string destinationDirec
             CopyDirectory(subDir.FullName, newDestinationDir);
         }
     }
+
+    public static bool IsSubdirectoryOf(string parentDirectory, string childDirectory)
+    {
+        var parentDirInfo = new DirectoryInfo(parentDirectory);
+        var childDirInfo = new DirectoryInfo(childDirectory);
+
+        while (childDirInfo.Parent is not null)
```

Comment:
I like this more thorough implementation, but would `childDirInfo.FullName.StartsWith(parentDirInfo.FullName);` work? This is me assuming DirectoryInfo normalized the paths.

Reply:
`.StartsWith()` doesn't work for things like `C:\temp` and `C:\temporary-documents\file.txt`.  In this case `.StartsWith` will return `true`, but it's not under the same path.  I considered ensuring a trailing directory separator, but that ends up with a bunch of edge cases that I didn't like.